scipy/stats/_qmc.py

```diff
@@ -1595,6 +1596,302 @@ def fast_forward(self, n: IntNumber) -> Sobol:
+    .. [2] `StackOverflow <https://stackoverflow.com/questions/66047540>`_.
+
+    Examples
+    --------
+    Generate a 2D sample using a `radius` of 0.2.
+
+    >>> import matplotlib.pyplot as plt
+    >>> from matplotlib.collections import PatchCollection
+    >>> from scipy.stats import qmc
+    >>>
+    >>> rng = np.random.default_rng()
+    >>> radius = 0.2
+    >>> engine = qmc.PoissonDisk(d=2, radius=0.2, seed=rng)
+    >>> sample = engine.random(20)
+
+    Visualizing the 2D sample and showing that no points are closer than
+    `radius`. ``radius/2`` is used to visualy have non-intersecting circle.
+    If two samples are exactly at `radius` from each other, then their circle
+    of radius ``radius/2`` will touch.
+
+    >>> fig, ax = plt.subplots()
+    >>> _ = ax.scatter(sample[:, 0], sample[:, 1])
+    >>> circles = [plt.Circle((xi, yi), radius=radius/2, fill=False)
+    ...            for xi, yi in sample]
+    >>> collection = PatchCollection(circles, match_original=True)
+    >>> ax.add_collection(collection)
+    >>> _ = ax.set(aspect='equal', xlabel=r'$x_1$', ylabel=r'$x_2$',
+    ...            xlim=[0, 1], ylim=[0, 1])
+    >>> plt.show()
+
+    Such visualization can be seen as circle packing: how many circle can
+    we put in the space. It is a np-hard problem. The method `fill_space`
+    can be used to add samples until no more samples can be added. This is
+    a hard problem parameters might need to be adjusted manually. Beware of
+    the dimension: as the dimensionality increases, the number of samples
+    required to fill the space increases exponentially
+    (curse-of-dimensionality).
+
+    """
+
+    def __init__(
+        self,
+        d: IntNumber,
+        *,
+        radius: DecimalNumber = 0.05,
+        hypersphere: Literal["volume", "surface"] = "volume",
+        candidates: IntNumber = 30,
+        seed: SeedType = None
+    ) -> None:
+        super().__init__(d=d, seed=seed)
+
+        hypersphere_sample = {
+            "volume": self._hypersphere_volume_sample,
+            "surface": self._hypersphere_surface_sample
+        }
+
+        try:
+            self.hypersphere_method = hypersphere_sample[hypersphere]
+        except KeyError as exc:
+            message = (
+                f"{hypersphere!r} is not a valid hypersphere sampling"
+                f" method. It must be one of {set(hypersphere_sample)!r}")
+            raise ValueError(message) from exc
+
+        # size of the sphere from which the samples are drawn relative to the
+        # size of a disk (radius)
+        # for the surface sampler, all new points are almost exactly 1 radius
+        # away from at least one existing sample +eps to avoid rejection
+        self.radius_factor = 2 if hypersphere == "volume" else 1.001
+        self.radius = radius
+        self.squared_radius = self.radius**2
+
+        # sample to generate per iteration in the hypersphere around center
+        self.candidates = candidates
```

Comment:
Do you mean just here or also the parameter name? I though it was simple like this. And since we have a parameter called `candidates` as a user I think I would not expect `self.candidates` to be something else than my parameter.

The other reason for not using `n_...` is that we are not consistent at all in terms of naming the number of something. So I didn't want to make a choice 😅

Reply:
I think `candidates` refer to the candidates themselves, whereas something like `nr_candidates` refers to, well, that. So just depends on what you mean with the name.

Reply:
I do mean a number of candidates. What about I mark it private and rename it to `self._nr_candidates` while keeping the name of the parameter as `candidates` which I think is clear especially since we have typing.

Reply:
Check the rest of SciPy? At a quick glance, it seems like elsewhere they also user nr when indicating number of.  See, e.g., https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.basinhopping.html#scipy.optimize.basinhopping

There will undoubtedly be many other examples.

Reply:
Ok then I will rename both to `ncandidates`